django/core/handlers/base.py

```diff
@@ -42,38 +44,47 @@ def load_middleware(self):
 
         Must be called after the environment is fixed (see __call__ in subclasses).
         """
+        self._request_middleware = []
         self._view_middleware = []
         self._template_response_middleware = []
         self._response_middleware = []
         self._exception_middleware = []
 
-        request_middleware = []
-        for middleware_path in settings.MIDDLEWARE_CLASSES:
-            mw_class = import_string(middleware_path)
-            try:
-                mw_instance = mw_class()
-            except MiddlewareNotUsed as exc:
-                if settings.DEBUG:
-                    if six.text_type(exc):
-                        logger.debug('MiddlewareNotUsed(%r): %s', middleware_path, exc)
-                    else:
-                        logger.debug('MiddlewareNotUsed: %r', middleware_path)
-                continue
-
-            if hasattr(mw_instance, 'process_request'):
-                request_middleware.append(mw_instance.process_request)
-            if hasattr(mw_instance, 'process_view'):
-                self._view_middleware.append(mw_instance.process_view)
-            if hasattr(mw_instance, 'process_template_response'):
-                self._template_response_middleware.insert(0, mw_instance.process_template_response)
-            if hasattr(mw_instance, 'process_response'):
-                self._response_middleware.insert(0, mw_instance.process_response)
-            if hasattr(mw_instance, 'process_exception'):
-                self._exception_middleware.insert(0, mw_instance.process_exception)
+        settings.MIDDLEWARE = settings.MIDDLEWARE_CLASSES
+        print settings.MIDDLEWARE
+        if settings.MIDDLEWARE is None:
+            handler = self._get_response_old
+            self._load_middleware_old()
+        else:
+            handler = self._get_response
+            for middleware_path in settings.MIDDLEWARE[::-1]:
```

Comment:
This is minor, but I'm curious -- any reason to use `[::-1]` over `reversed(settings.MIDDLEWARE)`? I think the latter is clearer, and if micro-optimization is a concern (shouldn't be here), I think it is better, as it just creates an iterator over the original list.